### PR TITLE
Refine naming for ext/datasource module 

### PR DESCRIPTION
**Pull request by luckyxiaoqiang**
### Describe what this PR does / why we need it

- Some renaming
- Refactor error handling

### Special notes for reviews

### CODE REVIEW

File: ext/datasource/helper.go [hunk 1]
```diff
@@ -151,9 +160,9 @@ func HotSpotParamRuleJsonArrayParser(src []byte) (interface{}, error) {
 	}
 
 	rules := make([]*hotspot.Rule, 0)
-	err := json.Unmarshal(src, &rules)
-	if err != nil {
-		return nil, err
+	if err := json.Unmarshal(src, &rules); err != nil {
+		desc := fmt.Sprintf("failed to convert source bytes to []*hotspot.Rule, err: %s", err.Error())
+		return nil, NewError(ConvertSourceError, desc)
 	}
```
Comment (louyuting) [inline ext/datasource/helper.go:166]: Maybe you should refine NewError() naming for HotSpotParamRulesUpdater

Comment (luckyxiaoqiang) [reply ext/datasource/helper.go:166]: Addressed.

**Review by louyuting — DISMISSED** [review]
Thanks for contribution

**Review by louyuting — DISMISSED** [review]
Thanks for contribution

**Review by louyuting — APPROVED** [review]
Thanks for contribution

**Pull request merged**